presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -198,6 +202,16 @@ public static Type typeConvert(Join.Type joinType)
         }
     }
 
+    public static boolean mustReplicate(JoinNode node, Lookup lookup)
+    {
+        return isScalar(lookup.resolve(node.getRight())) || node.isCrossJoin();
```

Comment:
it's shared between determineJoinDistributionType and ReorderJoins.  I don't want to remove the isScalar check because I don't want to introduce a regression for the non-stats case and it's a good heuristic anyway.